packages/shopify-codemod/transforms/iife-to-ternary-expression.js

```diff
@@ -0,0 +1,48 @@
+export default function iifeToTernaryExpression({source}, {jscodeshift: j}, {printOptions = {}}) {
+  function isImmediatelyInvokedFunctionExpression(expressionPath) {
+    let expression = expressionPath.node.expression
+    return expression.type === 'CallExpression' && expression.callee.type === 'ArrowFunctionExpression'
+  }
+
+  function containsOnlyIfStatement(functionNode) {
+    return nodeIsBlockStatementContainingOnly(functionNode.body, 'IfStatement');
+  }
+
+  function onlyReturnsOneOfTwoValues(ifStatementNode) {
+    return nodeIsBlockStatementContainingOnly(ifStatementNode.consequent, 'ReturnStatement') &&
+           nodeIsBlockStatementContainingOnly(ifStatementNode.alternate, 'ReturnStatement');
+  }
+
+  function nodeIsBlockStatementContainingOnly(node, childType) {
+    return node.type === 'BlockStatement' &&
+           node.body.length === 1 &&
+         node.body[0].type === childType;
+  }
+
+  function isConvertibleExpression(path) {
+    return isImmediatelyInvokedFunctionExpression(path) &&
+           containsOnlyIfStatement(path.node.expression.callee) &&
+           onlyReturnsOneOfTwoValues(path.node.expression.callee.body.body[0]);
+  }
+
+  const sourceAST = j(source);
+
+  sourceAST
+    .find(j.ExpressionStatement)
+    .filter(isConvertibleExpression)
+    .replaceWith(({node}) => {
+      let {expression: {callee: {body: functionBlockStatement}}} = node;
+    let {body: [{test,
```

Comment:
I'm not clever enough to unpack what this is doing.  Just from a plain, "what variables are being created here?" perspective.

On top of that, someone trying to understand a transform has to maintain a mental AST, and constantly compare it to the code.  Going crazy with destructuring (to me) moves the code away from a left-to-right traversal that makes relating the AST to code somewhat easy.

I'd go with something more explicit/verbose to help out the enfeebled:
```js
const ifStatement = path.get('expression', 'callee', 'body', 'body', '0');
const {test, consequent, alternate} = ifStatement.node;
```